modules/standard/BigInteger.chpl

```diff
@@ -765,194 +761,75 @@ module BigInteger {
   // Addition
   operator bigint.+(const ref a: bigint, const ref b: bigint): bigint {
     var c = new bigint();
-
-    const a_ = a.localize();
-    const b_ = b.localize();
-
-    mpz_add(c.mpz, a_.mpz, b_.mpz);
-
+    BigInteger.add(c, a, b);
     return c;
   }
 
-  operator bigint.+(const ref a: bigint, b: int): bigint {
+  operator bigint.+(const ref a: bigint, b): bigint
+  where isIntegralType(b.type) || isBoolType(b.type)
+  {
     var c = new bigint();
-    const a_ = a.localize();
-
-    if b >= 0 {
-      const b_ = b.safeCast(c_ulong);
-
-      mpz_add_ui(c.mpz, a_.mpz,  b_);
-
-    } else {
-      const b_ = (0 - b).safeCast(c_ulong);
-
-      mpz_sub_ui(c.mpz, a_.mpz, b_);
-    }
-
+    BigInteger.add(c, a, b);
     return c;
   }
 
-  operator bigint.+(a: int, const ref b: bigint): bigint {
+  operator bigint.+(a, const ref b: bigint): bigint
+  where isIntegralType(a.type) || isBoolType(a.type)
```

Comment:
Why are we checking if it is a bool type here?

Reply:
This was required because `apiTest.chpl:66` has tests for `+`, `-`, and `*` that take a `bool` as an argument. Based on #20125 and #20011, `integral` will not accept `bool` like an `int` will. So to match the behavior of the test it requires this extra hurdle.